lib/rspec/rails/example/view_example_group.rb

```diff
@@ -161,6 +163,12 @@ def _include_controller_helpers
           controller.controller_path = _controller_path
           controller.request.path_parameters[:controller] = _controller_path
           controller.request.path_parameters[:action]     = _inferred_action unless _inferred_action =~ /^_/
+          controller.request.path = ViewPathBuilder.new(::Rails.application.routes).path_for(controller.request.path_parameters)
+          ViewSpecMethods.add_to(::ActionView::TestCase::TestController)
+        end
+
+        after do
+          ViewSpecMethods.remove_from(::ActionView::TestCase::TestController)
```

Comment:
At this point, we're before any of the user's code, so they haven't set params yet. That means that when they do set params, later, we need to re-run the path creation. It felt odd to override `params#[]` to do that, so we opted to add new functionality. We can do the `params` thing instead if you'd rather.